lib/activerecord-import/import.rb

```diff
@@ -123,6 +123,15 @@ def supports_on_duplicate_key_update?
       false
     end
 
+    # returns true if the current database connection adapter
+    # supports setting the primary key of bulk imported models, otherwise
+    # returns false
+    def supports_setting_primary_key?
+      connection.supports_setting_primary_key?
+    rescue NoMethodError
```

Comment:
Oh man, I wish I could take back setting the precedent for rescuing `NoMethodError`. We don't need to change here, but this is when my present self wants to kick my past self. Should have relied on `respond_to?(...)` so we don't have to use exception for flow control here.